java/test/jmri/configurexml/LoadAndStoreTestBase.java

```diff
@@ -94,96 +105,180 @@ public LoadAndStoreTestBase(File file, boolean pass) {
         return SchemaTestBase.getDirectories(new File(directory, "load"), recurse, pass);
     }
 
-    static void checkFile(File inFile, File outFile) throws Exception {
-        // find comparison files
-        File compFile = new File(inFile.getCanonicalFile().getParentFile().getParent() + "/loadref/" + inFile.getName());
-        if (!compFile.exists()) {
-            compFile = inFile;
-        }
-        log.debug("   Chose comparison file " + compFile.getCanonicalPath());
-
+    public static void checkFile(File inFile1, File inFile2) throws Exception {
         // compare files, except for certain special lines
-        BufferedReader inFileStream = new BufferedReader(
-                new InputStreamReader(
-                        new FileInputStream(compFile)));
-        BufferedReader outFileStream = new BufferedReader(
-                new InputStreamReader(
-                        new FileInputStream(outFile)));
-        String inLine;
-        String outLine;
-        String nextIn;
-        String nextOut;
-        int count = 0;
-        inLine = inFileStream.readLine();
-        outLine = outFileStream.readLine();
-        while ( (nextIn = inFileStream.readLine()) != null && (nextOut = outFileStream.readLine()) != null) {
-            count++;
-
-            if (inLine.contains("<filehistory>") && outLine.contains("<filehistory>")) {
-                break;
+        BufferedReader fileStream1 = new BufferedReader(
+                new InputStreamReader(new FileInputStream(inFile1)));
+        BufferedReader fileStream2 = new BufferedReader(
+                new InputStreamReader(new FileInputStream(inFile2)));
+
+        String line1 = fileStream1.readLine();
+        String line2 = fileStream2.readLine();
+
+        int lineNumber1 = 0, lineNumber2 = 0;
+        String next1, next2;
+        while ((next1 = fileStream1.readLine()) != null && (next2 = fileStream2.readLine()) != null) {
+            lineNumber1++;
+            lineNumber2++;
+
+            // where the (empty) entryexitpairs line ends up seems to be non-deterministic…
```

Comment:
I'm not sure I understand the implications of this. Does this mean that an EntryExitPairsXml element gets created even if there are none to store or something else?

Reply:
>"an EntryExitPairsXml element gets created even if there are none to store

```
    public Element store(Object o) {
        EntryExitPairs p = (EntryExitPairs) o;
        Element element = new Element("entryexitpairs");  // NOI18N
        setStoreElementClass(element);
        ArrayList<LayoutEditor> editors = p.getSourcePanelList();
        if (editors.isEmpty()) {
            return element;
        }
```
If I change it to ```return null;``` here it eliminates the empty element… but I'm not familiar enough with these to know if it has any unintentional side effects.

Reply:
Lines 115-118 of jmri.jmrit.display.layoutEditor.configurexml.LayoutEditorXml explicitly check that the returned value of store() is not null before storing it:
```
                    Element e = jmri.configurexml.ConfigXmlManager.elementFromObject(sub);
                    if (e != null) {
                        panel.addContent(e);
                    }
```
So its safe to return null for empty collections.